python/src/uagents/context.py

```diff
@@ -585,136 +434,165 @@ async def send_raw_exchange_envelope(
-            session=session_id or uuid.uuid4(),
-            schema_digest=schema_digest,
+            session=self._session,
+            # schema_digest=message_schema_digest,
+            protocol=message_schema_digest,  # why use 'protocol'?
             protocol_digest=protocol_digest,
             expires=expires,
         )
-        env.encode_payload(json_message)
-        env.sign(sender)
-
-        headers = {"content-type": "application/json"}
-        if sync:
-            headers["x-uagents-connection"] = "sync"
-
-        for endpoint in endpoints:
-            try:
-                async with aiohttp.ClientSession() as session:
-                    async with session.post(
-                        endpoint,
-                        headers=headers,
-                        data=env.json(),
-                    ) as resp:
-                        success = resp.status == 200
-                        if success:
-                            if sync:
-                                return Envelope.parse_obj(await resp.json())
-
-                            return MsgStatus(
-                                status=DeliveryStatus.DELIVERED,
-                                detail="Message successfully delivered via HTTP",
-                                destination=destination_address,
-                                endpoint=endpoint,
-                            )
-                    log(
-                        logger,
-                        logging.WARNING,
-                        f"Failed to send message to {destination_address} @ {endpoint}: "
-                        + (await resp.text()),
-                    )
-            except aiohttp.ClientConnectorError as ex:
-                log(logger, logging.WARNING, f"Failed to connect to {endpoint}: {ex}")
-
-            except ValidationError as ex:
-                log(
-                    logger,
-                    logging.WARNING,
-                    f"Sync message to {destination} @ {endpoint} got invalid response: {ex}",
-                )
-
-            except Exception as ex:
-                log(
-                    logger,
-                    logging.WARNING,
-                    f"Failed to send message to {destination} @ {endpoint}: {ex}",
-                )
+        env.encode_payload(message_body)
+        env.sign(self.agent.sign_digest)
 
-        log(logger, logging.ERROR, f"Failed to deliver message to {destination}")
+        self._queue_envelope(env, sync)
 
         return MsgStatus(
-            status=DeliveryStatus.FAILED,
-            detail="Message delivery failed",
+            status=DeliveryStatus.QUEUED,
+            detail="Envelope successfully queued",
             destination=destination,
             endpoint="",
         )
+        # At this point we need the result of the send_raw_exchange_envelope
+        # but the result will be async so we need another way to access the
+        # return value
```

Comment:
Yes, indeed. How about adding a `Future` to self._outbound_messages` which the dispenser resolves to the corresponding `MsgStatus`, and then awaiting that here?